Makefile

```diff
@@ -463,21 +481,101 @@ site/src/api/typesGenerated.ts: scripts/apitypings/main.go $(shell find ./coders
 docs/admin/prometheus.md: scripts/metricsdocgen/main.go scripts/metricsdocgen/metrics
 	go run scripts/metricsdocgen/main.go
 	cd site
-	yarn run format:write ../docs/admin/prometheus.md
+	yarn run format:write:only ../docs/admin/prometheus.md
 
 coderd/apidoc/swagger.json: $(shell find ./scripts/apidocgen -not \( -path './scripts/apidocgen/node_modules' -prune \) -type f) $(wildcard coderd/*.go) $(wildcard codersdk/*.go)
 	./scripts/apidocgen/generate.sh
 	cd site
-	yarn run format:write ../docs/api ../docs/manifest.json ../coderd/apidoc/swagger.json
+	yarn run format:write:only ../docs/api ../docs/manifest.json ../coderd/apidoc/swagger.json
 
 update-golden-files: cli/testdata/.gen-golden
 .PHONY: update-golden-files
 
 cli/testdata/.gen-golden: $(wildcard cli/testdata/*.golden) $(GO_SRC_FILES)
-
 	go test ./cli -run=TestCommandHelp -update
 	touch "$@"
 
+# Ensure there's a .prettierrc.yaml project-wide when prettier is run outside
+# the site package.
+site/.prettierrc.yaml: .prettierrc.yaml Makefile
+	echo "#" > "$@"
+	echo "# Generated from ../$< by Makefile; DO NOT EDIT." >> "$@"
+	echo "#" >> "$@"
+
+	# Simplistic approach to rewriting override paths relative to site/.
+	isfile=0
+	IFS=$$'\n'
+	while read -r line; do
+		case "$$line" in
+			*files:*)
+				isfile=1
+				;;
+			*"- "*)
+				if [[ "$$isfile" == 1 ]]; then
+					if [[ "$${line}" = *'- "'* ]]; then
+						line=$${line//- \"/- \"..\/}
+					else
+						line=$${line//- /- ..\/}
+					fi
+					line=$${line//\/\//\/}
+					line=$${line//..\/site\//.\/}
+				fi
+				;;
+			*)
+				isfile=0
+				;;
+		esac
+		echo "$$line" >> "$@"
+	done < "$<"
```

Comment:
Could this not be done with sed or something instead? Or using `yq` as well if you really need to only change the values within `files:`?

Reply:
I can give `yq` a try 👍🏻 (I don't think we have a dependency on it yet, though). Multiline in sed is kind of unwieldy, I can't imagine it would be robust/future-proof.

Reply:
Due to v1 dependency on `yq` v3 (I used v4), I jumped through some hoops: [coder/coder@`9cf9a44` (#5505)](https://github.com/coder/coder/pull/5505/commits/9cf9a44faac71c0d51f80c406fdaa82e5cdd461a). Added the dependency to various places in [coder/coder@`f612e75` (#5505)](https://github.com/coder/coder/pull/5505/commits/f612e75e32a8850c5fd995cb9f364dc00aeb97c3) (ignore whitespace).